src/common/file-utils.cpp

```diff
@@ -34,21 +34,136 @@
 #include <stdio.h>
 #include "uniwin.h"
 #include <errno.h>
+#else
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#endif        // !_MSC_VER
+#ifdef _WIN32 // for rnp_mkstemp
 #include <locale>
 #include <codecvt>
 #include <random>
+#endif
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
+#include <rnp/rnp_sdk.h>
+#ifdef _WIN32
+#include <locale>
+#include <codecvt>
+#endif
+
+extern "C" {
+int
+rnp_unlink(const char *filename)
+{
+#ifdef _WIN32
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8conv;
+    std::wstring filename_w = utf8conv.from_bytes(filename);
+    return _wunlink(filename_w.c_str());
 #else
-#include <sys/stat.h>
-#endif // _MSC_VER
+    return unlink(filename);
+#endif
+}
 
 bool
 rnp_file_exists(const char *path)
 {
     struct stat st;
-    return stat(path, &st) == 0 && S_ISREG(st.st_mode);
+    return rnp_stat(path, &st) == 0 && S_ISREG(st.st_mode);
+}
+
+bool
+rnp_dir_exists(const char *path)
+{
+    struct stat st;
+    return rnp_stat(path, &st) == 0 && S_ISDIR(st.st_mode);
+}
+
+bool
+rnp_path_exists(const char *path)
+{
+    struct stat st;
+    return rnp_stat(path, &st) == 0;
+}
+}
+
+int
+rnp_open(const char *filename, int oflag, int pmode)
+{
+#ifdef _WIN32
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8conv;
+    std::wstring filename_w = utf8conv.from_bytes(filename);
+    return _wopen(filename_w.c_str(), oflag, pmode);
+#else
+    return open(filename, oflag, pmode);
+#endif
+}
+
+FILE *
+rnp_fopen(const char *filename, const char *mode)
+{
+#ifdef _WIN32
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8conv;
+    std::wstring filename_w = utf8conv.from_bytes(filename);
+    std::wstring mode_w = utf8conv.from_bytes(mode);
+    return _wfopen(filename_w.c_str(), mode_w.c_str());
+#else
+    return fopen(filename, mode);
+#endif
+}
+
+int
+rnp_stat(const char *filename, struct stat *statbuf)
+{
+#ifdef _WIN32
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8conv;
+    std::wstring filename_w = utf8conv.from_bytes(filename);
```

Comment:
This and backward conversion snippet is used multiple times over the code, what about moving it to `std::wstring  wstr_from_utf8(...)`/`std::string wstr_to_utf8(...)`?

Reply:
Moved to `str-utils`